src/main/java/org/folio/circulation/infrastructure/storage/CheckOutLockRepository.java

```diff
@@ -0,0 +1,85 @@
+package org.folio.circulation.infrastructure.storage;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.Environment;
+import org.folio.circulation.domain.CheckOutLock;
+import org.folio.circulation.domain.LoanAndRelatedRecords;
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.CollectionResourceClient;
+import org.folio.circulation.support.http.client.Response;
+import org.folio.circulation.support.http.client.ResponseInterpreter;
+import org.folio.circulation.support.results.Result;
+
+import java.lang.invoke.MethodHandles;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+
+import static org.folio.circulation.support.http.ResponseMapping.forwardOnFailure;
+import static org.folio.circulation.support.http.ResponseMapping.mapUsingJson;
+
+public class CheckOutLockRepository {
+  private static final Logger log = LogManager.getLogger(MethodHandles.lookup().lookupClass());
+  private final CollectionResourceClient checkOutLockClient;
+  private final List<Integer> retryIntervals;
+  private final Vertx vertx = Vertx.currentContext() != null ? Vertx.currentContext().owner() : Vertx.vertx();
+
+  public CheckOutLockRepository(Clients clients, List<Integer> retryIntervals) {
+    this.checkOutLockClient = clients.checkOutLockClient();
+    this.retryIntervals = retryIntervals;
+  }
+
+  public void createLockWithRetry(int noOfAttempts, CompletableFuture<CheckOutLock> future, LoanAndRelatedRecords records) {
```

Comment:
I have used vertx.setTimer() method to delay the retry process. The setTimer method is executing in a separate thread. So our current thread is not waiting for the response of this method. Since I want my current thread to wait until this method completion, I am passing a future and completes it. Now, The caller method(acquireLock) from checkOutByBarcodeResource will wait until this entire retry completes.